src/hostapi/pulseaudio/pa_linux_pulseaudio_cb.c

```diff
@@ -119,21 +119,23 @@ void PaPulseAudio_ReleaseOperation(PaPulseAudio_HostApiRepresentation *hostapi,
 
     while( waitOperation > 0 )
     {
-        PaPulseAudio_Lock( hostapi->mainloop );
-        pa_threaded_mainloop_wait( hostapi->mainloop );
-        PaPulseAudio_UnLock( hostapi->mainloop );
 
+        PaPulseAudio_Lock( hostapi->mainloop );
         localOperationState = pa_operation_get_state( localOperation );
 
-        // No wait if operation have been DONE or CANCELLED
-        if( localOperationState != PA_OPERATION_RUNNING)
+        if( localOperationState == PA_OPERATION_RUNNING )
+        {
+            pa_threaded_mainloop_wait( hostapi->mainloop );
```

Comment:
It looks scary to calla  wait() function under a lock because of the danger of deadlocks.
But this follows the PulseAudio example code. So it seems OK.